src/hooks/use-database.js

```diff
@@ -0,0 +1,9 @@
+// @flow
+import { useContext } from 'react'
+import { DatabaseContext } from '../DatabaseProvider'
+import type Database from '../Database'
+
+export function useDatabase(): Database {
+  const database = useContext(DatabaseContext)
```

Comment:
hmm, the signature here is different than that of TS signature. I think `useDatabase` SHOULD return Database. And if it's not there, I think it should throw an error, not just return undefined.